flax/core/lift.py

```diff
@@ -52,7 +52,8 @@ def _dedup_scopes(scopes):
       path.append(scope.name)
       scope = scope.parent
     if max_parent is not leaf:
-      del minimal_set[leaf]
+      if leaf in minimal_set:
```

Comment:
This was altered to support duplicate leaf scopes that occur in some edge-cases with top-level module adoption.

Reply:
Thanks, duplicate leaves are supposed to be supported but probably forgot to test.